buildSrc/subprojects/integration-testing/src/main/kotlin/gradlebuild/distribution-testing.gradle.kts

```diff
@@ -16,15 +16,25 @@
 package gradlebuild
 
 import gradlebuild.cleanup.extension.CleanupExtension
+import gradlebuild.cleanup.services.CachesCleaner
 import gradlebuild.integrationtests.tasks.DistributionTest
 
 plugins {
     java
+    id("gradlebuild.module-identity")
+}
+
+val intTestHomeDir = rootProject.layout.projectDirectory.dir("intTestHomeDir")
+
+val cachesCleanerService = gradle.sharedServices.registerIfAbsent("cachesCleaner", CachesCleaner::class) {
+    parameters.homeDir.set(intTestHomeDir)
```

Comment:
For some reason, I can not pass a `GradleVersion` as a parameter to build service - it fails to serialize. While at the same time it works when configuring a task.

Reply:
`GradleVersion` is not serialisable. It worked before because it was not declared as an input to the task.